gver-dsl/src/test/java/org/dmfs/gver/dsl/git/GitVersionTest.java

```diff
@@ -202,4 +204,43 @@ void testDirtyStaged(String bundle, String version)
                                 )))));
     }
 
+
+    Assertion illegal_commit_fails = withResources("",
+        new TempDir(),
+        new Repository(getClass().getClassLoader().getResource("0.1.0-alpha.bundle"), "main"),
+
+        // TODO: replace with initialized resource see https://github.com/saynotobugsorg/confidence/issues/70
+        (tempDir, repo) -> withResource(() -> {
+                new File(tempDir, "newFile").createNewFile();
+                Git git = new Git(repo);
+                git.add().addFilepattern("newFile").call();
+                git.commit().setMessage("commit #invalid").call();
+
+                return new WithResource.Resource<org.eclipse.jgit.lib.Repository>()
+                {
+                    @Override
+                    public void close()
+                    {
+                        repo.close();
+                    }
+
+                    @Override
+                    public org.eclipse.jgit.lib.Repository value()
+                    {
+
+                        return repo;
+                    }
+                };
+            },
+            r -> assertionThat(new GitVersion(new FirstOf(
+                    ChangeType.INVALID.when(new CommitMessage(new Contains("#invalid"))),
+                    ChangeType.MAJOR.when(new CommitMessage(new Contains("#major"))),
+                    ChangeType.MINOR.when(new CommitMessage(new Contains("#minor"))),
+                    ChangeType.PATCH.when(new CommitMessage(new Contains("#patch"))),
+                    ChangeType.NONE.when(new CommitMessage(new Contains("#trivial"))),
+                    ChangeType.UNKNOWN.when(((treeWalk, commit, branches) -> true))),
+                    new Suffixes(), ignored -> "alpha"),
+                has("version", v -> () -> v.value(repo), is(throwing(IllegalArgumentException.class))))));
```

Comment:
@sschuberth , at present, this is how you would test if a task throws an exception.